beluga_amcl/src/amcl_node.cpp

```diff
@@ -500,7 +422,7 @@ void AmclNode::map_callback(nav_msgs::msg::OccupancyGrid::SharedPtr map) {
   enable_tf_broadcast_ = false;
 }
 
-void AmclNode::timer_callback() {
+void AmclNode::do_periodic_timer_callback() {
```

Comment:
@serraramiro1 meta: why can't this method body live in `BaseAMCLNode`? Same for the `initial_pose_callback` for instance. Is it the `particle_filter_` type? Would CRTP fix this? Or perhaps just another virtual method.

Reply:
I thought about this a bit and I couldn't find a solution that wouldn't hurt readability.
I want to keep these common features state and measurement agnostic, and that makes it a little bit hard.

I know CRTP could accomplish this but personally I feel like top level modules should be easy to follow in terms of how data flows.

WDYT?

Reply:
I did extract the common subscription into the base class though...

Reply:
> I know CRTP could accomplish this but personally I feel like top level modules should be easy to follow in terms of how data flows.

Well, if you look at Beluga that ship has sailed a very long time ago. But this is a general comment and I don't mind that strongly if readability is good. We can generalize further when and if the time comes.